django/test/runner.py

```diff
@@ -515,6 +521,13 @@ def __init__(self, pattern=None, top_level=None, verbosity=1,
                 for pattern in test_name_patterns
             }
 
+        if self.verbosity == 0:
+            self.level = NOTSET
+        elif self.verbosity == 1:
+            self.level = INFO
+        else:
+            self.level = DEBUG
```

Comment:
You shouldn't modify `DiscoverRunner.__init__()`. All of the logic can go in `DiscoverRunner.log()`. In addition to not being necessary, a problem with setting `self.level` is that there will now no longer be one source of truth. For example, a caller that sets `DiscoverRunner.verbosity` after instantiating the runner will get incorrect results when calling `DiscoverRunner.log()`.

Reply:
Sure. I have removed all the modifications in the __init__ and was planning on updating the defination of the `log` method as,
```
def log(self, msg, level = None):
        if level is None:
            level = logging.INFO
        
        if self.verbosity == 0:
            logging_level = logging.NOTSET
        elif self.verbosity == 1:
            logging_level = logging.INFO
        else:
            logging_level = logging.DEBUG

        if logging_level >= level:
            print(msg)
```